ui/components/telemetry/grafana/GrafanaComponent.js

```diff
@@ -212,31 +216,43 @@ class GrafanaComponent extends Component {
 
   handleChange = (name) => (data) => {
     if (name === 'grafanaURL' && !!data) {
-      this.setState({ urlError: false });
-      this.setState({ [name]: data.value });
+      const grafanaURL = data?.value;
+      const isInvalidURL = !(
+        grafanaURL.toLowerCase().startsWith('http://') ||
+        grafanaURL.toLowerCase().startsWith('https://')
+      );
+
+      if (isInvalidURL) {
+        this.setState({ urlError: true, [name]: '' });
+        return;
+      }
+
+      this.setState({ urlError: false, [name]: grafanaURL });
     }
 
-    const grafanaConnectionObj = data.target?.value;
     if (name === 'grafanaBoardSearch') {
       if (this.boardSearchTimeout) clearTimeout(this.boardSearchTimeout);
       this.boardSearchTimeout = setTimeout(() => getGrafanaBoards(this), 500); // to delay the search by a few.
     }
 
-    getCredentialByID(grafanaConnectionObj?.credential_id).then((res) => {
-      const grafanaCfg = {
-        grafanaURL: grafanaConnectionObj?.value || '',
-        grafanaAPIKey: res?.secret?.secret || '',
-        grafanaBoardSearch:
-          grafanaConnectionObj?.metadata?.grafanaBoardSearch || this.state?.grafanaBoardSearch,
-        grafanaBoards: grafanaConnectionObj?.metadata['grafana_boards'] || [],
-        selectedBoardsConfigs: grafanaConnectionObj?.metadata['selectedBoardsConfigs'],
-        connectionID: grafanaConnectionObj?.id,
-        connectionName: grafanaConnectionObj?.name,
-      };
-      this.props.updateGrafanaConfig({ grafana: grafanaCfg });
-    });
-
-    // this.setState({ [name]: event.target.value });
+    const grafanaCfg = {
+      grafanaURL: data?.value || '',
+      grafanaAPIKey: '',
+      grafanaBoardSearch: data?.metadata?.grafanaBoardSearch || this.state?.grafanaBoardSearch,
+      grafanaBoards: data?.metadata?.grafana_boards || [],
+      selectedBoardsConfigs: data?.metadata?.selectedBoardsConfigs || [],
+      connectionID: data?.id,
+      connectionName: data?.name,
+    };
+
+    if (typeof data?.credential_id !== 'undefined') {
+      getCredentialByID(data?.credential_id).then((res) => {
```

Comment:
@shanukun this seems a concern here. Can you verify that?
The ui currently expecting to get the details of api key from the `secret` value of credentials api but currently we don't have that way to add that secret so you would probably get this value as undefined.

Reply:
Users are to supply the API key, if their Grafana needs it.

Reply:
When creating a new connection, users provide the API key. However, if a connection has already been established, the only option to acquire the API key is to use the getCredentialByID function. The credential_id may be obtained from a grafana config request (`/api/telemetry/metrics/grafana/config`). 

However, after looking into the Board Search textfield and handleChange closely, I think this function needs to be called right after retrieving the connection config.